src/components/partnership/partnership.service.ts

```diff
@@ -603,4 +628,67 @@ export class PartnershipService {
       );
     }
   }
+
+  protected async isFirstPartnership(projectId: string): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: projectId }),
+        relation('out', '', 'partnership', { active: true }),
+        node('partnership'),
+      ])
+      .return(['partnership'])
+      .asResult<{ partnership?: Node }>()
+      .first();
+
+    return result?.partnership ? false : true;
+  }
+
+  /**
+   *
+   * match current primary partnership, its project, and all other partnerships
+   * set current to primary false
+   * set first of other partnerships to primary true
+   */
+  protected async movePrimaryPartnership(
+    direction: 'to' | 'from',
+    partnershipId: string
+  ): Promise<void> {
+    const createdAt = DateTime.local();
+    const newPropertyNodeProps = {
+      createdAt,
+      value: direction === 'from',
+      sortValue: determineSortValue(direction === 'from'),
+    };
+    const query = this.db
+      .query()
+      .match([
+        node('partnership', 'Partnership', { id: partnershipId }),
+        relation('in', '', 'partnership', { active: true }),
+        node('project', 'Project'),
+        relation('out', '', 'partnership', { active: true }),
+        node('otherPartnership'),
```

Comment:
How do we know that `otherPartnership` doesn't match the current `partnership` as well?

Reply:
I think we just need a condition here so that `otherPartnership` matches the opposite value. That way we have both partnerships that need their primary property flipped.

Reply:
```
MATCH (partnership:Partnership { id: 'id' })<-[:partnership { active: true }]-(project:Project)-[:partnership { active: true }]->(otherPartnership)
```
This always returns all partnerships attached to `project` except selected `partnership`. FYI, I cannot find a reference doc for this tricky query.

Reply:
I tested how Neo4j handles this kind of query by creating the following nodes with this cypher:
`create (:child {name: "child1"})<-[:toChild]-(:mother {name: "mom"})-[:toChild]->(:child{name:"child2"})`
I then queried using this:
`match (:child {name:"child1"})<-[:toChild]-(:mother)-[:toChild]->(child) return child.name`
and got 1 match: `"child1"`
and this query:
`match (:mother)-[:toChild]->(child) return child.name`
got 2 nodes returned: `"child1", "child2"`
I believe its safe to leave off a condition, but adding it will make developer intent clear.